src/spritecache.cpp

```diff
@@ -43,17 +43,8 @@ static inline SpriteCache *GetSpriteCache(uint index)
 
 SpriteCache *AllocateSpriteCache(uint index)
 {
-	if (index >= _spritecache_items) {
-		/* Add another 1024 items to the 'pool' */
-		uint items = Align(index + 1, 1024);
-
-		Debug(sprite, 4, "Increasing sprite cache to {} items ({} bytes)", items, items * sizeof(*_spritecache));
-
-		_spritecache = ReallocT(_spritecache, items);
-
-		/* Reset the new items and update the count */
-		memset(_spritecache + _spritecache_items, 0, (items - _spritecache_items) * sizeof(*_spritecache));
-		_spritecache_items = items;
+	if ((size_t)index >= _spritecache.size()) {
```

Comment:
/me browses through PR

Finds this one place where there is still a C-cast

Takes a stick and points.

Look.

Here :P


(sorry :D)